contracts/core/Controller.sol

```diff
@@ -719,6 +707,9 @@ contract Controller is Initializable, OwnableUpgradeSafe, ReentrancyGuardUpgrade
         // store new vault
         accountVaultCounter[_args.owner] = vaultId;
         vaultType[_args.owner][vaultId] = _args.vaultType;
+        vaults[_args.owner][vaultId].setMarginPool(
+            poolV2.whitelistedRibbonVault(_args.owner) ? address(poolV2) : address(pool)
+        );
 
```

Comment:
when a vault is created, we immediately associate it with a margin pool in the [Vault](https://github.com/ribbon-finance/avocado/blob/master/contracts/libs/MarginVault.sol#L33) struct. We determine which margin pool to set by checking if owner of vault is a whitelisted address from the margin pool v2 ad-hoc mapping. Addresses of balancer, badger, spell vaults (and others) will be whitelisted. Retail vaults will not be and as a result will be directed to vanilla margin pool like before